aws/middleware/endpoints_test.go

```diff
@@ -0,0 +1,195 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+type mockSerializeHandler func(ctx context.Context, in middleware.SerializeInput) (out middleware.SerializeOutput, metadata middleware.Metadata, err error)
+
+func (m mockSerializeHandler) HandleSerialize(
+	ctx context.Context, in middleware.SerializeInput,
+) (out middleware.SerializeOutput, metadata middleware.Metadata, err error) {
+	return m(ctx, in)
+}
+
+func TestResolveServiceEndpoint(t *testing.T) {
+	initContext := func(signingName, endpointID, region string) context.Context {
+		ctx := context.Background()
+		ctx = SetSigningName(ctx, signingName)
+		ctx = setEndpointID(ctx, endpointID)
+		ctx = setRegion(ctx, region)
+		return ctx
+	}
+
+	cases := []struct {
```

Comment:
Nit it would be great to use map[string]struct here so these test cases can be named. helps as description and debugging